### PR TITLE
Fix Issue 19871 - Copy constructor rejects valid code if default construction is disabled

### PR DESCRIPTION
**Pull request by RazvanN7**
The initial idea was that when the compiler searches for copy constructors it should stop when it has found one, however, it should also check if the user has defined an rvalue normal constructor. The patch makes it so that the search for copy constructors does not stop when a copy constructor was found.

### CODE REVIEW

File: test/fail_compilation/failCopyCtor.d [hunk 1]
```diff
@@ -3,7 +3,7 @@ TEST_OUTPUT:
 ---
 fail_compilation/failCopyCtor.d(10): Error: `struct A` may not define both a rvalue constructor and a copy constructor
 fail_compilation/failCopyCtor.d(12):        rvalue constructor defined here
-fail_compilation/failCopyCtor.d(13):        copy constructor defined here
+fail_compilation/failCopyCtor.d(14):        copy constructor defined here
```
Comment (jacob-carlborg) [inline test/fail_compilation/failCopyCtor.d]: Why was this changed?

Comment (RazvanN7) [reply test/fail_compilation/failCopyCtor.d]: I thought it doesn't matter if the last or first copy constructor is chosen, but you are right, it should be the first.

Comment (RazvanN7) [reply test/fail_compilation/failCopyCtor.d]: Fixed

**Review by thewilsonator — APPROVED** [review]
